ViewHolder margin fixes

zsmb13 opened the pull request:
### Description

Various fixes around ViewHolder margins and similar things:

- Fix start/end margins on ViewHolders
- Use `tools` namespace in ViewHolder layouts for guideline positioning, as setting those values does nothing (overridden at runtime by a decorator)
- Hides the delivery status on Giphy ViewHolders
- Fix incorrect spacing of ViewHolder content when avatars are hidden

| Before | After |
| --- | --- |
| ![Screenshot_1608039968](https://user-images.githubusercontent.com/12054216/102981934-00c78f80-450a-11eb-917b-22bc79f22220.png) | ![Screenshot_1608040391](https://user-images.githubusercontent.com/12054216/102993361-56596780-451d-11eb-9774-6fe2e90a37fc.png) |

### Checklist

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] PR targets the `develop` branch
- [ ] ~Changelog updated with client-facing changes~
- [ ] ~New code is covered by unit tests~
- [x] Comparison screenshots added for visual changes
- [x] Reviewers added

ogkuzmin commented:
I see in design there is some margin after avatar. We don't have it. Maybe it worths to fix it here?
<img width="369" alt="image" src="https://user-images.githubusercontent.com/11807573/102984482-b43a7f00-451e-11eb-9e46-03b82f723015.png">

zsmb13 commented:
Updated this PR: added the margins @ogkuzmin asked about, simplified the way `GravityDecorator` works (so that more of the original XML constraints are respected, instead of completely overriding them).
